x/clock/abci.go

```diff
@@ -31,12 +33,19 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 		childCtx := ctx.WithGasMeter(sdk.NewGasMeter(p.ContractGasLimit))
 		_, err = k.GetContractKeeper().Sudo(childCtx, contract, message)
 		if err != nil {
+			errorExists = true
 			errorExecs[idx] = addr
-			continue
+
+			// Jail contract on error
+			err = k.JailContract(ctx, addr)
+			if err != nil {
+				log.Printf("[x/clock] Failed to Error Contract %s: %v", addr, err)
+			}
 		}
 	}
 
-	if len(errorExecs) > 0 {
+	// Log errors if present
+	if errorExists {
```

Comment:
what does this solve compared to the previous implementation `len(errorExecs) > 0`

Reply:
The length of the array was always greater than 0, so the statement was always true.